javascript/ql/src/semmle/javascript/security/dataflow/ExceptionXss.qll

```diff
@@ -0,0 +1,87 @@
+/**
+ * Provides a taint-tracking configuration for reasoning about cross-site 
+ * scripting vulnerabilities where the taint-flow passes through a thrown 
+ * exception. 
+ */
+
+import javascript
+
+module ExceptionXss {
+  import DomBasedXssCustomizations::DomBasedXss as DomBasedXssCustom
+  import ReflectedXssCustomizations::ReflectedXss as ReflectedXssCustom 
+  import Xss::DomBasedXss as DomBasedXss 
+  import Xss::ReflectedXss as ReflectedXSS
+  import Xss::StoredXss as StoredXss
+  import Xss as XSS
+
+  DataFlow::ExceptionalInvocationReturnNode getCallerExceptionalReturn(Function func) {
+    exists(DataFlow::InvokeNode call |
+      not call.isImprecise() and
+      func = call.(DataFlow::InvokeNode).getACallee() and
+      result = call.getExceptionalReturn()
+    )
+  }
+
+  DataFlow::Node getExceptionalSuccssor(DataFlow::Node pred) {
+    if exists(getEnclosingTryStmt(pred.asExpr().getEnclosingStmt()))
+    then
+      result.(DataFlow::ParameterNode).getParameter() = getEnclosingTryStmt(pred
+              .asExpr()
+              .getEnclosingStmt()).getACatchClause().getAParameter()
+    else result = getCallerExceptionalReturn(pred.getContainer())
+  }
+
+  predicate canThrowSensitiveInformation(DataFlow::Node node) {
+    // i have to do this "dual" check because there are two data-flow nodes associated with reflective calls.
+    node = any(DataFlow::InvokeNode call).getAnArgument() and
+    not node = any(DataFlow::InvokeNode call | exists(call.getACallee())).getAnArgument()
+    or
+    node.asExpr().getEnclosingStmt() instanceof ThrowStmt
+  }
+
+  TryStmt getEnclosingTryStmt(Stmt stmt) {
+    stmt.getParentStmt+() = result.getBody() and
+    not exists(TryStmt mid |
+      stmt.getParentStmt+() = mid.getBody() and mid.getParentStmt+() = result.getBody()
+    )
+  }
+  
+  class NotYetThrown extends DataFlow::FlowLabel {
```

Comment:
Could you document the semantics of this flow label? From its use below it isn't entirely clear to me.